Userland/Libraries/LibJS/Tests/builtins/Number/Number.js

```diff
@@ -29,6 +29,42 @@ test("constructor without new", () => {
     expect(Number("foo")).toBeNaN();
     expect(Number("10e10000")).toBe(Infinity);
     expect(Number("-10e10000")).toBe(-Infinity);
+    expect(Number("0b1")).toBe(1);
+    expect(Number("0B1")).toBe(1);
+    expect(Number("0b01")).toBe(1);
+    expect(Number("0b11")).toBe(3);
+    expect(Number("0b")).toBeNaN();
+    expect(Number("0B")).toBeNaN();
+    expect(Number("-0b1")).toBeNaN();
+    expect(Number("+0b1")).toBeNaN();
+    expect(Number("0b1.1")).toBeNaN();
+    expect(Number("0b1e10")).toBeNaN();
+    expect(Number("0b1e+10")).toBeNaN();
+    expect(Number("0b1e-10")).toBeNaN();
+    expect(Number("0o7")).toBe(7);
+    expect(Number("0O7")).toBe(7);
+    expect(Number("0o07")).toBe(7);
+    expect(Number("0o77")).toBe(63);
+    expect(Number("0o")).toBeNaN();
+    expect(Number("0O")).toBeNaN();
+    expect(Number("-0o1")).toBeNaN();
+    expect(Number("+0o1")).toBeNaN();
+    expect(Number("0o1.1")).toBeNaN();
+    expect(Number("0o1e10")).toBeNaN();
+    expect(Number("0o1e+10")).toBeNaN();
+    expect(Number("0o1e-10")).toBeNaN();
+    expect(Number("0x1")).toBe(1);
+    expect(Number("0X1")).toBe(1);
+    expect(Number("0x01")).toBe(1);
+    expect(Number("0x11")).toBe(17);
+    expect(Number("0x")).toBeNaN();
+    expect(Number("0X")).toBeNaN();
+    expect(Number("-0x1")).toBeNaN();
+    expect(Number("+0x1")).toBeNaN();
+    expect(Number("0x1.1")).toBeNaN();
+    expect(Number("0x1e10")).toBe(7696);
+    expect(Number("0x1e+10")).toBeNaN();
+    expect(Number("0x1e-10")).toBeNaN();
```

Comment:
:+1: Great tests!

Perhaps we could add a couple more like `"0x1_2"` and `"1_2"`.
Since the spec says 
> A [StringNumericLiteral](https://tc39.es/ecma262/#prod-StringNumericLiteral) cannot include a [NumericLiteralSeparator](https://tc39.es/ecma262/#prod-NumericLiteralSeparator).

In the note under https://tc39.es/ecma262/#prod-StringNumericLiteral